packages/lodestar/src/chain/regen/queued.ts

```diff
@@ -52,72 +53,125 @@ export class QueuedStateRegenerator implements IStateRegenerator {
-    const state = this.stateCache.get(parentBlock.stateRoot);
-    if (state) {
-      return state;
-    }
 
-    // The state is not immediately available in the caches, enqueue the job
-    return this.jobQueue.push({key: "getPreState", args: [block]});
+      // The state is not immediately available in the caches, enqueue the job
+      return this.jobQueue.push({key: "getPreState", args: [block, rCaller]});
+    } catch (e) {
+      this.metrics?.regenFnTotalErrors.inc({caller, entrypoint: RegenFnName.getPreState});
+      throw e;
+    }
   }
 
-  async getCheckpointState(cp: phase0.Checkpoint): Promise<CachedBeaconState<allForks.BeaconState>> {
-    // First attempt to fetch the state from cache before queueing
-    const checkpointState = this.checkpointStateCache.get(cp);
-    if (checkpointState) {
-      return checkpointState;
+  async getCheckpointState(
+    cp: phase0.Checkpoint,
+    rCaller: IRegenCaller
+  ): Promise<CachedBeaconState<allForks.BeaconState>> {
+    const {caller} = rCaller || {};
+    this.metrics?.regenFnCallTotal.inc({caller, entrypoint: RegenFnName.getCheckpointState});
+    const rmetrics = this.regen.getRegenFnMetrics({caller, entrypoint: RegenFnName.getCheckpointState});
+    try {
+      // First attempt to fetch the state from cache before queueing
+      const checkpointState = this.checkpointStateCache.get(cp, rmetrics);
+      if (checkpointState) {
+        return checkpointState;
+      }
+      // The state is not immediately available in the cache, enqueue the job
+      return this.jobQueue.push({key: "getCheckpointState", args: [cp, rCaller]});
+    } catch (e) {
+      this.metrics?.regenFnTotalErrors.inc({caller, entrypoint: RegenFnName.getCheckpointState});
+      throw e;
     }
-    // The state is not immediately available in the cache, enqueue the job
-    return this.jobQueue.push({key: "getCheckpointState", args: [cp]});
   }
 
-  async getBlockSlotState(blockRoot: Root, slot: Slot): Promise<CachedBeaconState<allForks.BeaconState>> {
-    return this.jobQueue.push({key: "getBlockSlotState", args: [blockRoot, slot]});
+  async getBlockSlotState(
+    blockRoot: Root,
+    slot: Slot,
+    rCaller: IRegenCaller
+  ): Promise<CachedBeaconState<allForks.BeaconState>> {
+    const {caller} = rCaller || {};
+    this.metrics?.regenFnCallTotal.inc({caller, entrypoint: RegenFnName.getBlockSlotState});
+    return this.jobQueue.push({key: "getBlockSlotState", args: [blockRoot, slot, rCaller]});
   }
 
-  async getState(stateRoot: Root): Promise<CachedBeaconState<allForks.BeaconState>> {
-    // First attempt to fetch the state from cache before queueing
-    const state = this.stateCache.get(stateRoot);
-    if (state) {
-      return state;
+  async getState(stateRoot: Root, rCaller: IRegenCaller): Promise<CachedBeaconState<allForks.BeaconState>> {
+    const {caller} = rCaller || {};
+    this.metrics?.regenFnCallTotal.inc({caller, entrypoint: RegenFnName.getState});
```

Comment:
I would differentiate between total calls that get here and calls that get queued. At this level I would just metric `regenFnCallTotal`, not errors, do that on jobQueueProcessor. Then at the top of jobQueueProcessor count `regenFnCallQueueTotal`. Then we can compute the ratio between those to see the cache hit vs cache miss.

Reply:
understood, do we need 'regenFnCallQueueTotal'? `regenFnCallDuration` is a histogram and gives a count along with it to do this ratio.